runhouse/resources/hardware/utils.py

```diff
@@ -238,22 +294,110 @@ def rsync(
         *,
         up: bool,
         # Advanced options.
-        filter_options: Optional[str] = None,
+        filter_options: Optional[str] = None,  # RH MODIFIED
         log_path: str = os.devnull,
         stream_logs: bool = True,
         max_retry: int = 1,
+        return_cmd: bool = False,
     ) -> None:
-        # temp update rsync filters to exclude docs, when syncing over runhouse folder
-        org_rsync_filter = cr.RSYNC_FILTER_OPTION
-        if filter_options:
-            cr.RSYNC_FILTER_OPTION += f" --filter='{filter_options}'"
-        super().rsync(
-            source,
-            target,
-            up=up,
-            log_path=log_path,
-            stream_logs=stream_logs,
-            max_retry=max_retry,
+        """Uses 'rsync' to sync 'source' to 'target'.
```

Comment:
I think there's functionality in .run() that makes it hard to directly reuse skypilot code without also porting it over, but think for this one if it's just filter options we can maybe leave it as is? If skypilot modifies this function on their end, I assume we'd generally want to automatically inherit those changes too and not have to reduplicate updates/have two sources of truth.

Reply:
Even though we don't use the filter options anymore, we still modify this one with the return_cmd argument, which we need for pexpect. It's not ideal but I don't see a way around it. We can also try just contributing this option directly up to SkyPilot.